Src/FluentAssertions/DataTableCollectionAssertionExtensions.cs

```diff
@@ -0,0 +1,189 @@
+                        assertion.Subject);
+            }
+
+            return new AndConstraint<GenericCollectionAssertions<DataTable>>(assertion);
+        }
+
+        /// <summary>
+        /// Asserts that an object reference refers to a different object than another object reference refers to.
+        /// </summary>
+        /// <param name="unexpected">The unexpected object</param>
+        /// <param name="because">
+        /// A formatted phrase as is supported by <see cref="string.Format(string,object[])" /> explaining why the assertion
+        /// is needed. If the phrase does not start with the word <i>because</i>, it is prepended automatically.
+        /// </param>
+        /// <param name="becauseArgs">
+        /// Zero or more objects to format using the placeholders in <paramref name="because"/>.
+        /// </param>
+        public static AndConstraint<GenericCollectionAssertions<DataTable>> NotBeSameAs(
+            this GenericCollectionAssertions<DataTable> assertion, DataTableCollection unexpected, string because = "",
+            params object[] becauseArgs)
+        {
+            if (assertion.Subject is NonGenericCollectionWrapper<DataTableCollection, DataTable> wrapper)
+            {
+                var actualSubject = wrapper.UnderlyingCollection;
+
+                Execute.Assertion
+                    .UsingLineBreaks
+                    .ForCondition(!ReferenceEquals(actualSubject, unexpected))
+                    .BecauseOf(because, becauseArgs)
+                    .FailWith("Did not expect {context:table collection} to refer to {0}{reason}.", unexpected);
+            }
+
+            return new AndConstraint<GenericCollectionAssertions<DataTable>>(assertion);
+        }
+
+        /// <summary>
+        /// Assert that the current collection of <see cref="DataTable"/>s has the same number of tables as
+        /// <paramref name="otherDataSet" />.
+        /// </summary>
+        /// <param name="otherDataSet">The other <see cref="DataSet"/> with the same expected number of tables</param>
+        /// <param name="because">
+        /// A formatted phrase as is supported by <see cref="string.Format(string,object[])" /> explaining why the assertion
+        /// is needed. If the phrase does not start with the word <i>because</i>, it is prepended automatically.
+        /// </param>
+        /// <param name="becauseArgs">
+        /// Zero or more objects to format using the placeholders in <paramref name="because" />.
+        /// </param>
+        public static AndConstraint<GenericCollectionAssertions<DataTable>> HaveSameCount(
+            this GenericCollectionAssertions<DataTable> assertion, DataSet otherDataSet, string because = "",
+            params object[] becauseArgs)
+        {
+            return assertion.HaveSameCount(otherDataSet.Tables, because, becauseArgs);
```

Comment:
This line is uncovered

Reply:
Ah, that makes sense. I'll go add tests that cover these. :-)